core/src/main/java/org/mule/runtime/core/api/component/Component.java

```diff
@@ -7,18 +7,22 @@
 package org.mule.runtime.core.api.component;
 
 import static org.mule.runtime.core.api.processor.ReactiveProcessor.ProcessingType.BLOCKING;
+
 import org.mule.runtime.core.api.Event;
 import org.mule.runtime.core.api.construct.FlowConstructAware;
-import org.mule.runtime.core.api.processor.Processor;
 import org.mule.runtime.core.api.management.stats.ComponentStatistics;
+import org.mule.runtime.core.api.processor.Processor;
 
 /**
  * A <code>Component</code> component processes a {@link Event} by invoking the component instance that has been configured,
  * optionally returning a result.
  * <p/>
  * Implementations of <code>Component</code> can use different types of component implementation, implement component instance
  * pooling or implement <em>bindings</em> which allow for service composition.
+ * 
+ * @deprecated Transport infrastructure is deprecated.
  */
+@Deprecated
 public interface Component extends Processor, FlowConstructAware {
```

Comment:
Not without some other refactoring. It is used in `org.mule.runtime.core.internal.lifecycle.phases.MuleContext*Phase` to configure the lifecycle of the components in the registry (same as `LegacyConnector`).
I created https://www.mulesoft.org/jira/browse/MULE-13023 to track this.